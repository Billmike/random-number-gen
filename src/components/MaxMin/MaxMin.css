.minimax-container {
  position: absolute;
  top: 20%;
  left: 25px;
  font-family: monospace;
  font-size: 14px;
}

.max {
  margin-bottom: 15px;
}
