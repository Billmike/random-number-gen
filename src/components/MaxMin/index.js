import React from 'react';
import './MaxMin.css';

const MaxMin = (props) => {
  const phoneNumbers = props.numbers.length > 0 ? props.numbers.map(phoneNumber => phoneNumber.userPhoneNumber) : 0;
  const maxNumber = phoneNumbers === 0 ? 0 : Math.max(...phoneNumbers);
  const minNumber = phoneNumbers === 0 ? 0 : Math.min(...phoneNumbers);
  return (
    <div className="minimax-container">
      <div className="max">
        The maximum generated Phone Number: {phoneNumbers === 0 ? '000' : `090${maxNumber}`}
      </div>
      <div>
        The minimum generated Phone Number: {phoneNumbers === 0 ? '000' : `090${minNumber}`}
      </div>
    </div>
  )
}

export default MaxMin;
