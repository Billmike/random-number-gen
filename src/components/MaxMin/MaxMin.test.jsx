import React from 'react';
import { shallow } from 'enzyme';
import MaxMin from './index';

test('<MaxMin /> renders correctly', () => {
  const props = {
    numbers: []
  }

  const wrapper = shallow(<MaxMin {...props} />);
  expect(wrapper).toMatchSnapshot();
});

test('<MaxMin />', () => {
  const props = {
    numbers: [
      {userPhoneNumber: 1234566},
      {userPhoneNumber: 986945}
    ]
  }
  const wrapper = shallow(<MaxMin {...props} />);
  expect(wrapper).toMatchSnapshot()
})
