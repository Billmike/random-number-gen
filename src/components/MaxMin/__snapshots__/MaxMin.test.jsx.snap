// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaxMin /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MaxMin
    numbers={
      Array [
        Object {
          "userPhoneNumber": 1234566,
        },
        Object {
          "userPhoneNumber": 986945,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="max"
        >
          The maximum generated Phone Number: 
          0901234566
        </div>,
        <div>
          The minimum generated Phone Number: 
          090986945
        </div>,
      ],
      "className": "minimax-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "The maximum generated Phone Number: ",
            "0901234566",
          ],
          "className": "max",
        },
        "ref": null,
        "rendered": Array [
          "The maximum generated Phone Number: ",
          "0901234566",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "The minimum generated Phone Number: ",
            "090986945",
          ],
        },
        "ref": null,
        "rendered": Array [
          "The minimum generated Phone Number: ",
          "090986945",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="max"
          >
            The maximum generated Phone Number: 
            0901234566
          </div>,
          <div>
            The minimum generated Phone Number: 
            090986945
          </div>,
        ],
        "className": "minimax-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "The maximum generated Phone Number: ",
              "0901234566",
            ],
            "className": "max",
          },
          "ref": null,
          "rendered": Array [
            "The maximum generated Phone Number: ",
            "0901234566",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "The minimum generated Phone Number: ",
              "090986945",
            ],
          },
          "ref": null,
          "rendered": Array [
            "The minimum generated Phone Number: ",
            "090986945",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<MaxMin /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MaxMin
    numbers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="max"
        >
          The maximum generated Phone Number: 
          000
        </div>,
        <div>
          The minimum generated Phone Number: 
          000
        </div>,
      ],
      "className": "minimax-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "The maximum generated Phone Number: ",
            "000",
          ],
          "className": "max",
        },
        "ref": null,
        "rendered": Array [
          "The maximum generated Phone Number: ",
          "000",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "The minimum generated Phone Number: ",
            "000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "The minimum generated Phone Number: ",
          "000",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="max"
          >
            The maximum generated Phone Number: 
            000
          </div>,
          <div>
            The minimum generated Phone Number: 
            000
          </div>,
        ],
        "className": "minimax-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "The maximum generated Phone Number: ",
              "000",
            ],
            "className": "max",
          },
          "ref": null,
          "rendered": Array [
            "The maximum generated Phone Number: ",
            "000",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "The minimum generated Phone Number: ",
              "000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "The minimum generated Phone Number: ",
            "000",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
